src/nvim/drawscreen.c

```diff
@@ -165,14 +165,6 @@ bool default_grid_alloc(void)
   // size is wrong.
 
   grid_alloc(&default_grid, Rows, Columns, true, true);
-  StlClickDefinition *new_tab_page_click_defs =
-    xcalloc((size_t)Columns, sizeof(*new_tab_page_click_defs));
-
-  stl_clear_click_defs(tab_page_click_defs, tab_page_click_defs_size);
-  xfree(tab_page_click_defs);
-
-  tab_page_click_defs = new_tab_page_click_defs;
-  tab_page_click_defs_size = Columns;
```

Comment:
Moving this to `redraw_tabline()` may require adding some `NULL` checks in `mouse.c`.

Reply:
Hmm, can any of the `tab_page_click_defs` references in `mouse.c` can be accessed before `redraw_tabline()`?

Reply:
If anything, I guess something like this could cause problems, but it works fine:

    nvim -u NONE src/nvim/mouse.c +"tabedit % | lua vim.api.nvim_input_mouse('left', 'press','', 0, 0, 0)"

Reply:
Run the following script on startup:
```vim
set showtabline=2
call nvim_input_mouse('move', '', '', 0, 0, 0)
call feedkeys("\<LeftMouse>", 'tx')
```